website/static/js/share/utils.js

```diff
@@ -190,30 +234,253 @@ utils.removeFilter = function(vm, filter){
     utils.search(vm);
 };
 
-utils.arrayEqual = function(a, b) {
+/* Tests array equality */
+utils.arrayEqual = function (a, b) {
     return $(a).not(b).length === 0 && $(b).not(a).length === 0;
 };
 
-utils.loadStats = function(vm){
-    vm.statsLoaded(false);
-
+/* Loads the raw and normalized data for a specific result */
+utils.loadRawNormalized = function(result){
+    var source = encodeURIComponent(result.shareProperties.source);
+    var docID = encodeURIComponent(result.shareProperties.docID);
     return m.request({
         method: 'GET',
-        url: '/api/v1/share/stats/?' + $.param({q: utils.buildQuery(vm)}),
-        background: true
-    }).then(function(data) {
-        vm.statsData = data;
-        $.map(Object.keys(vm.graphs), function(type) {
-            if(type === 'shareDonutGraph') {
-                var count = data.charts.shareDonutGraph.columns.filter(function(val){return val[1] > 0;}).length;
-                $('.c3-chart-arcs-title').text(count + ' Provider' + (count !== 1 ? 's' : ''));
+        url: 'api/v1/share/documents/' + source + '/' + docID + '/',
+        unwrapSuccess: function(data) {
+            var unwrapped = {};
+            var normed = JSON.parse(data.normalized);
+            var allRaw = JSON.parse(data.raw);
+            unwrapped.normalized = JSON.parse(data.normalized);
+            unwrapped.raw = allRaw.doc;
+            unwrapped.rawfiletype = allRaw.filetype;
+            unwrapped.normalized = normed;
+
+            return unwrapped;
+        },
+        unwrapError: function(data) {
+            var error = {};
+            error.rawfiletype = 'json';
```

Comment:
In the case of 500-series errors, you should probably log to Sentry and display an error message with the OSF support email.